File: Kymdan/src/main/java/com/kymdan/backend/services/loai_san_pham/LoaiSanPhamService.java
package com.kymdan.backend.services.loai_san_pham;

import com.kymdan.backend.entity.LoaiSanPham;
import com.kymdan.backend.model.ThongBaoDTO;
import com.kymdan.backend.model.LoaiSanPhamDTO;

import java.util.List;

public interface LoaiSanPhamService {
    List<LoaiSanPham> xemTatCa();

    List<?> locTheoGia(long mucGia);

    LoaiSanPham timBangMaLoai(String maLoai);

    LoaiSanPham timBangTen(String ten);

    ThongBaoDTO taoMoi(LoaiSanPhamDTO loaiSanPhamDTO);

    ThongBaoDTO sua(LoaiSanPhamDTO loaiSanPhamDTO);

    ThongBaoDTO xoa(String ten);

    List<LoaiSanPham> xemLoaiMoi();

    List<LoaiSanPham> xemLoaiBanChay();

    List<LoaiSanPham> timTheoTen(String ten);
}

File: Kymdan_front_end/src/app/service/loai-san-pham.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class LoaiSanPhamService {
  public readonly API: string = 'http://localhost:8080/loaiSanPham';

  constructor(
    public http: HttpClient
  ) {
  }

  xemTatCa(): Observable<any> {
    return this.http.get(this.API + '/xemTatCa');
  }

  locTheoGia(gia): Observable<any> {
    return this.http.get(this.API + '/locTheoGia/' + gia);
  }

  timBangMaLoai(maLoai): Observable<any> {
    return this.http.get(this.API + '/timBangMaLoai/' + maLoai);
  }

  taoMoi(loaiSanPham): Observable<any> {
    return this.http.post(this.API + '/taoMoi', loaiSanPham, {headers: {skip: 'true'}});
  }

  sua(loaiSanPham): Observable<any> {
    return this.http.post(this.API + '/sua', loaiSanPham, {headers: {skip: 'true'}});
  }

  xoa(ten): Observable<any> {
    return this.http.get(this.API + '/xoa/' + ten);
  }

  xemLoaiMoi(): Observable<any> {
    return this.http.get(this.API + '/xemLoaiMoi');
  }

  xemLoaiBanChay(): Observable<any> {
    return this.http.get(this.API + '/xemLoaiBanChay');
  }

  timTheoTen(ten): Observable<any> {
    return this.http.get(this.API + '/timTheoTen/' + ten);
  }
}
